src/cobra/io/web/load.py

```diff
@@ -0,0 +1,201 @@
+
+
+import gzip
+import logging
+import pathlib
+from typing import TYPE_CHECKING, Iterable, Optional, Union
+
+import appdirs
+import diskcache
+import httpx
+import libsbml
+
+from ..sbml import _sbml_to_model
+from .abstract_model_repository import AbstractModelRepository
+from .bigg_models_repository import BiGGModels
+from .biomodels_repository import BioModels
+
+
+if TYPE_CHECKING:
+    from cobra.core import Model
+
+
+logger = logging.getLogger(__name__)
+
+
+def load_model(
+    model_id: str,
+    repositories: Iterable[AbstractModelRepository] = (BiGGModels(), BioModels()),
+    cache: bool = True,
+    cache_directory: Optional[Union[pathlib.Path, str]] = None,
+) -> "Model":
+    """
+    Download an SBML model from a remote repository.
+
+    Downloaded SBML documents are by default stored in a cache on disk such that future
+    access is much faster. By default, models can be loaded from the following
+    repositories:
+
+    * BiGG Models
+    * BioModels
+
+    You can use the ``AbstractModelRepository`` class as a parent to implement your own
+    repository accessor which you pass to the ``load_model`` function. In case you
+    implement a new interface, please consider submitting a pull request to COBRApy.
+
+    Parameters
+    ----------
+    model_id : str
+        The identifier of the desired metabolic model. This is typically repository
+        specific.
+    repositories : iterable, optional
+        An iterable of repository accessor instances. The model_id is searched in order.
+    cache : bool, optional
+        Whether or not to use the local caching mechanism (default yes).
+    cache_directory : pathlib.Path or str, optional
+        A path where the cache should reside if caching is desired. The default
+        directory depends on the operating system.
+
+    Returns
+    -------
+    Model
+        A model instance generated from the SBML document.
+
+    Raises
+    ------
+    RuntimeError
+        As with any internet connection, there are multiple errors that can occur.
+
+    Examples
+    --------
+    # Most of the time calling `load_model` with an identifier should be enough.
+    >>> print(load_model("e_coli_core"))
+    e_coli_core
+    >>> print(load_model("MODEL1510010000"))
+    MODEL1510010000
+
+    See Also
+    --------
+    BiGGModels
+    BioModels
+
+    """
+    if cache:
+        if cache_directory is None:
+            cache_directory = pathlib.Path(
+                appdirs.user_cache_dir(appname="cobrapy", appauthor="opencobra")
+            )
+        data = _cached_load(
+            model_id=model_id,
+            repositories=repositories,
+            cache_directory=pathlib.Path(cache_directory),
+        )
+    else:
+        data = _fetch_model(model_id=model_id, repositories=repositories)
+    return get_model_from_gzip_sbml(data)
+
+
+def _cached_load(
+    model_id: str,
+    repositories: Iterable[AbstractModelRepository],
+    cache_directory: pathlib.Path,
```

Comment:
This together with the maximum cache dir size and a time when to clean it up should be set in the configuration and not here.

Reply:
Yes, good idea.